File: cpp/051-060/Jump Game.cpp
class Solution {
public:
    bool canJump(vector<int>& nums) {
        
        int maxpos(0);

        for (int i = 0; i < nums.size(); i++) {
            if (i <= maxpos) {
                int tmp = i + nums[i];
                if(tmp >= (nums.size()-1)) return true;
                maxpos = max(tmp, maxpos);
            }
            else return false;
        }
        return false;
    }
    
};

File: python/051-060/Jump Game.py
class Solution:
    # @param {integer[]} nums
    # @return {boolean}
    def canJump(self, nums):
        l = len(nums)
        res = 0
        for i in range(l):
            if i <= res:
                res = max(res, i+nums[i]);
        if res >= l-1:
            return True
        return False